### PR TITLE
PLT-6226 Fixing races with licensing

### PR DESCRIPTION
**Pull request by coreyhulen**
Merging/Building was giving me some trouble so I've resubmitted this PR.

Enterprise PR at https://github.com/mattermost/enterprise/pull/181

### CODE REVIEW

**Comment by ccbrown** [comment]
Actually, I see now that `SetLicense(nil)` actually sets the license to a default struct. In that case, why do we even do things like `if utils.IsLicensed() && *utils.License().Features.LDAP` as opposed to just `if *utils.License().Features.LDAP`?

File: utils/license.go [hunk 1]
```diff
@@ -49,23 +76,36 @@ func LoadLicense(licenseBytes []byte) {
...
+		SetIsLicensed(false)
+		license = &model.License{
+			Features: new(model.Features),
+		}
+		license.Features.SetDefaults()
+		licenseValue.Store(license)
 
-	return false
+		SetClientLicense(map[string]string{"IsLicensed": "false"})
+		//clientLicenseValue.Store(getClientLicense(License()))
```
Comment (ccbrown) [inline utils/license.go]: Should this commented line be here?

**Comment by coreyhulen** [comment]
This PR is the first step in cleaning up the Licensing stuff not necessarily the final output.

I tried not to change any of the implementation.  If you notice the original initialization was

```
-var License *model.License = &model.License{
 -	Features: new(model.Features),
 -}
```

So it was set.  I added `SetLicense(nil)` as a short had for this.  Also nil was only ever set from the unit tests.

File: utils/license.go (hunk 1 above)
Comment (coreyhulen) [inline utils/license.go]: I removed it.

**Pull request merged**